### PR TITLE
Add GPU support.

### PR DESCRIPTION
**Pull request by jdoliner**

### CODE REVIEW

File: src/client/pps/pps.proto [hunk 1]
```diff
@@ -112,6 +112,9 @@ message ResourceSpec {
   // The amount of memory, in bytes, each worker needs (in bytes, with allowed
   // SI suffixes (M, K, G, Mi, Ki, Gi, etc).
   string memory = 2;
+
+  // The number of GPUs each worker needs.
+  int64 gpu = 3;
```
Comment (msteffen) [inline src/client/pps/pps.proto]: Drive-by comment: Does k8s support partial GPU sharing? If so, should we make this a float?

Comment (jdoliner) [inline src/client/pps/pps.proto]: Drive by commenting is always welcome :D, currently it does not support partial GPUs and it's unclear if that's in the cards, I didn't see any indication that it was.

**Comment by jdoliner** [comment]
Actually we can kind of have an integration test for this, I added a test that creates a pipeline with a GPU request and checks to make sure it propagates to k8s. The pod won't actually schedule because the machine doesn't have a gpu so it's not a real test for gpus but it tests things up to and including the k8s api which is something.
